tephra-core/src/main/java/co/cask/tephra/distributed/ElasticPool.java

```diff
@@ -107,23 +112,23 @@ public ElasticPool(int sizeLimit) {
    *
    * @return an element
    */
-  public T obtain() throws E {
-    T element = getOrCreate();
+  public T obtain() throws E, TimeoutException {
     while (true) {
```

Comment:
That's for retry on InterruptedException.
Alternatively, we can propagate the InterruptedException.